### DISCUSSION TITLE
Cross Compile Beats using mage

### DISCUSSION THREAD

**Discussion by serak** [magefile/mage#415] [Q&A] [answered]
```
C:\Users\Serak\Desktop\goProject\beats>mage -compile -goos=aix -goarch=ppc64
cmd/go: unsupported GOOS/GOARCH pair windows/ppc64
Error: error compiling magefiles
```
what am i doing wrong here ? go already supports AIX on PPC64 architecture.

**Comment by natefinch** [comment, marked as answer]
Ahh, you hit a really weird edge case in the UX.

Compile requires a name to give the output binary. When you run that command line, the stdlib flags package thinks that you want the output binary to be called `-goos=aix` ... and thus doesn't treat that as a separate flag getting set. So, mage doesn't know you set -goos=aix at all.

so it should be `mage -compile=beats_magefile -goos=aix -goarch=ppc64`

It took me a while to understand this, because I, too, forgot that compile needs a filename, and the fact that the flags package will take the next string regardless of the fact that it starts with a dash is a little unintuitive.
